src/coreclr/vm/tailcallhelp.cpp

```diff
@@ -579,7 +601,20 @@ MethodDesc* TailCallHelp::CreateCallTargetStub(const TailCallInfo& info)
 
         pCode->EmitLDARG(ARG_RET_VAL);
         pCode->EmitLDLOC(resultLcl);
-        EmitStoreTyHnd(pCode, info.RetTyHnd);
+        bool isSigned;
+        if (NeedsReturnWidening(info.RetTyHnd, &isSigned))
+        {
+            if (isSigned)
+                pCode->EmitCONV_I4();
+            else
+                pCode->EmitCONV_U4();
```

Comment:
Is this even needed?

Reply:
It should not be. `EmitLDLOC` should do the appropriate widening implicitly.